src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -76,7 +72,7 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
-            _ = this.startContextFetcher.TryFetch(payload, out HttpContext context);
+            HttpContext context = payload as DefaultHttpContext;
```

Comment:
I don't think this is guaranteed to always be of type `DefaultHttpContext`. Might be a bit of an unusual case but I share this example here https://github.com/open-telemetry/opentelemetry-dotnet/pull/3498#discussion_r932771470. `DefaultHttpContext` is derived from `HttpContext`.

Reply:
Not sure if I understood. With the example (adding HttpContextFactory), I am getting the payload as DefaultHttpContext type.

Reply:
Oh yea you're right, that example uses the DefaultHttpContextFactory. My presumption is that you could use a different factory that might generate a totally different type.

Reply:
Not a fully working example, but something like 

```csharp
private class TestHttpContextFactory : IHttpContextFactory
 {
    public HttpContext Create(IFeatureCollection featureCollection)
    {
        return new MyHttpContext();
    }
}

private class MyHttpContext : HttpContext
{
}
```

Reply:
changed to HttpContext.